drivers/ethernet/eth_stm32_hal.c

```diff
@@ -891,7 +891,21 @@ static struct eth_stm32_hal_dev_data eth0_data = {
 		.Instance = (ETH_TypeDef *)DT_INST_REG_ADDR(0),
 		.Init = {
 #if !defined(CONFIG_SOC_SERIES_STM32H7X)
+#if defined(CONFIG_ETH_STM32_AUTO_NEGOTIATION_ENABLE)
```

Comment:
As I understand the auto-negotiation should not define duplex and speed mode if it is enabled: https://en.m.wikipedia.org/wiki/Autonegotiation 

Therefore I would prefer to split auto-negotiation, duplex mode and speed onto separate if-def's.

Reply:
@Nukersson I think what is proposed is what you request. Would you mind having a new look ?

Reply:
You're right. Sorry. Updated my review.